datafusion/optimizer/src/simplify_expressions/expr_simplifier.rs

```diff
@@ -2320,16 +2337,10 @@ mod tests {
         assert_no_change(regex_match(col("c1"), lit("f_o")));
 
         // empty cases
-        assert_change(regex_match(col("c1"), lit("")), like(col("c1"), "%"));
```

Comment:
🤔 the original rewrite rule looks incorrect to me (as in the orignal rule shouldn't be rewriting regexp match). We can fix this in some other PR

```sql
postgres=# select regexp_match('foo', '');
 regexp_match
--------------
 {""}
(1 row)

postgres=# select like('foo', '%');
 like
------
 t
(1 row)
```